tests/test_categories.py

```diff
@@ -60,4 +69,32 @@ def test_invalid_path():
     with pytest.raises(Exception):
         cc_client.update_categories('file', target=invalid_categories[[0]])
     with pytest.raises(Exception):
-        cc_client.update_categories('file', target=invalid_categories[[1]])
\ No newline at end of file
+        cc_client.update_categories('file', target=invalid_categories[[1]])
+
+def test_update_account_categories_with_list():
+    """Test that should get error while updating account categories with target as a list type"""
+    cc_client = CashCtrlClient()
+    with pytest.raises(ValueError):
+        cc_client.update_categories('account', target=categories)
+
+def test_update_file_categories_with_dict():
+    """Test that should get error while updating file categories with target as a dict type"""
+    cc_client = CashCtrlClient()
+    with pytest.raises(ValueError):
+        cc_client.update_categories('file', target=account_categories)
+
+def test_account_category_update():
+    """Test that new updates categories for accounts and then restores initial state"""
+    cc_client = CashCtrlClient()
+    initial_categories = cc_client.list_categories('account')
+    initial_paths = { row['path']: row['number'] for _, row in initial_categories.iterrows() }
+    cc_client.update_categories('account', target=account_categories)
+    remote_categories = cc_client.list_categories('account')
+    assert set(account_categories).issubset(remote_categories['path']), (
```

Comment:
I do not think a set operation is sufficient to verify that category numbers are correct.

(Also applies for the test further below)